qa/rolling-upgrade/build.gradle

```diff
@@ -29,112 +30,102 @@ task bwcTest {
   group = 'verification'
 }
 
-for (Version version : bwcVersions.wireCompatible) {
+configurations {
+  restSpec
+}
+
+dependencies {
+  restSpec project(':rest-api-spec')
+}
+
+processTestResources {
+  from ({ zipTree(configurations.restSpec.singleFile) }) {
+    include 'rest-api-spec/api/**'
+  }
+  dependsOn configurations.restSpec
+}
+
+for (Version bwcVersion : bwcVersions.wireCompatible) {
   /*
    * The goal here is to:
    * <ul>
    *  <li>start three nodes on the old version
    *  <li>run tests with systemProperty 'tests.rest.suite', 'old_cluster'
-   *  <li>shut down one node
-   *  <li>start a node with the new version
+   *  <li>upgrade one node
    *  <li>run tests with systemProperty 'tests.rest.suite', 'mixed_cluster'
-   *  <li>shut down one node on the old version
-   *  <li>start a node with the new version
+   *  <li>upgrade one more node
    *  <li>run tests with systemProperty 'tests.rest.suite', 'mixed_cluster' again
-   *  <li>shut down the last node with the old version
-   *  <li>start a node with the new version
+   *  <li>updgrade the last node
    *  <li>run tests with systemProperty 'tests.rest.suite', 'upgraded_cluster'
-   *  <li>shut down the entire cluster
    * </ul>
-   *
-   * Be careful: gradle dry run spits out tasks in the wrong order but,
-   * strangely, running the tasks works properly.
    */
-  String baseName = "v${version}"
+  String baseName = "v${bwcVersion}"
 
-  Task oldClusterTest = tasks.create(name: "${baseName}#oldClusterTest", type: RestIntegTestTask) {
-    mustRunAfter(precommit)
-  }
+  testClusters {
+    "${baseName}" {
+      versions = [ bwcVersion.toString(), project.version ]
+      numberOfNodes = 3
 
-  configure(extensions.findByName("${baseName}#oldClusterTestCluster")) {
-    bwcVersion = version
-    numBwcNodes = 3
-    numNodes = 3
-    clusterName = 'rolling-upgrade'
-    setting 'repositories.url.allowed_urls', 'http://snapshot.test*'
-    if (version.onOrAfter('5.3.0')) {
+      setting 'repositories.url.allowed_urls', 'http://snapshot.test*'
+      setting 'path.repo', "${buildDir}/cluster/shared/repo/${baseName}"
       setting 'http.content_type.required', 'true'
+      javaHome = project.file(project.ext.runtimeJavaHome)
```

Comment:
Isn't this the default? And if not, should it be?

Reply:
I wouldn't want to have a plugin rely on some project extension for a default.
I plan to deal with as part of #40531